flutter-idea/src/io/flutter/run/common/CommonTestConfigUtils.java

```diff
@@ -242,18 +242,8 @@ private void populateTestTypeMap(@NotNull FlutterOutline outline, @NotNull PsiFi
    * <p>
    * Used to ensure that we don't get stuck with out-of-date line markers.
    */
-  private static class LineMarkerUpdatingListener implements ActiveEditorsOutlineService.Listener {
-    @NotNull final CommonTestConfigUtils commonTestConfigUtils;
-    @NotNull final Project project;
-    @NotNull final ActiveEditorsOutlineService service;
-
-    private LineMarkerUpdatingListener(@NotNull CommonTestConfigUtils commonTestConfigUtils,
-                                       @NotNull Project project,
-                                       @NotNull ActiveEditorsOutlineService service) {
-      this.commonTestConfigUtils = commonTestConfigUtils;
-      this.project = project;
-      this.service = service;
-    }
+  private record LineMarkerUpdatingListener(@NotNull CommonTestConfigUtils commonTestConfigUtils, @NotNull Project project,
+                                            @NotNull ActiveEditorsOutlineService service) implements ActiveEditorsOutlineService.Listener {
```

Comment:
Another win. Note too how we can de-duplicate the double specification of `@NotNull` annotations.